ctru-rs/src/lib.rs

```diff
@@ -11,20 +11,27 @@ pub fn init() {
 
     use std::panic::PanicInfo;
 
+    let main_thread = thread::current().id();
+
     // Panic Hook setup
     let default_hook = std::panic::take_hook();
     let new_hook = Box::new(move |info: &PanicInfo| {
-        println!("\x1b[1;31m\n--------------------------------------------------");
+        println!("\x1b[1;31m\n--------------------------------------------------"); // Red ANSI Color Code
         default_hook(info);
-        println!("\nPress SELECT to exit the software");
-        let hid = services::hid::Hid::init().unwrap();
 
-        loop {
-            hid.scan_input();
-            if hid.keys_down().contains(services::hid::KeyPad::KEY_SELECT) {
-                break;
+        // Only for panics in the main thread
+        if main_thread == thread::current().id() && console::console_exists() {
+            println!("\nPress SELECT to exit the software");
+            let hid = services::hid::Hid::init().unwrap();
+
+            loop {
+                hid.scan_input();
+                if hid.keys_down().contains(services::hid::KeyPad::KEY_SELECT) {
+                    break;
+                }
```

Comment:
We could insert a yield here if we wanted to allow other threads to continue. I think there's a function from libctru or libc that would work?

Reply:
Threads with higher priority will continue automatically. A thread with lower priority probably depends on `main`, so at that point we may just cause a deadlock. I think it's fine this way.